File: VB/ChartDesignerCustomization/MainForm.vb
Imports DevExpress.XtraCharts.Designer

Namespace CrosshairOptions
    Partial Public Class Form1
        Inherits DevExpress.XtraBars.Ribbon.RibbonForm

        Public Sub New()
            InitializeComponent()

        End Sub

        Private Sub barButtonItem1_ItemClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs) Handles barButtonItem1.ItemClick
'            #Region "#DesignerCustomization"
            Dim designer As New ChartDesigner(chartControl1)
            designer.AddElementMenuOptions.ShowAddSeriesMenuItem = False
            designer.AddElementMenuOptions.ShowAddSeriesTitleMenuItem = False
            designer.Caption = "Chart Designer"
            designer.EnableLargeDataSetWarning = True

            AddHandler designer.ChartElementHighlighting, AddressOf OnChartElementHighlighting
            AddHandler designer.ChartElementSelecting, AddressOf OnChartElementSelecting
            AddHandler designer.ChartStructureUpdating, AddressOf OnChartStructureUpdating

            designer.ShowDialog()
'            #End Region ' #DesignerCustomization
        End Sub

        #Region "#BehaviorEvents"
        ' The ChartStructureUpdating event allows for modifying chart model.
        Private Sub OnChartStructureUpdating(ByVal sender As Object, ByVal e As ChartStructureChangingEventArgs)
            e.ChartModel.Series.AllowAddChild = False
            For Each seriesModel As SeriesModel In e.ChartModel.Series
                seriesModel.AllowChangeVisibility = False
                seriesModel.AllowRemove = False
            Next seriesModel
            e.ChartModel.Legends.ShowInStructureControl = False
        End Sub

        ' The ChartElementSelecting event allows for preventing element selection or
        ' modifying the selecting element edit controls.
        Private Sub OnChartElementSelecting(ByVal sender As Object, ByVal e As ChartElementSelectingEventArgs)
            If TypeOf e.ElementModel Is SeriesBaseModel Then
                e.Cancel = True
                Return
            End If
            If TypeOf e.ElementModel Is LegendModel Then
                e.ShowPropertiesTab = False
                e.ShowDataTab = False
                e.CustomOptionsControl = New CustomLegendOptionsControl(e.ElementModel)
            End If
        End Sub

        ' The ChartElementSelecting event allows for preventing element highlight.
        Private Sub OnChartElementHighlighting(ByVal sender As Object, ByVal e As ChartElementHighlightingEventArgs)
            Dim series As SeriesBaseModel = TryCast(e.ElementModel, SeriesBaseModel)
            If series Is Nothing Then
                e.Cancel = True
                Return
            End If

        End Sub
        #End Region ' #BehaviorEvents
    End Class
End Namespace

File: CS/ChartDesignerCustomization/MainForm.cs
using DevExpress.XtraCharts.Designer;

namespace CrosshairOptions {
    public partial class Form1 : DevExpress.XtraBars.Ribbon.RibbonForm {
        public Form1() {
            InitializeComponent();

        }

        private void barButtonItem1_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            #region #DesignerCustomization
            ChartDesigner designer = new ChartDesigner(chartControl1);
            designer.AddElementMenuOptions.ShowAddSeriesMenuItem = false;
            designer.AddElementMenuOptions.ShowAddSeriesTitleMenuItem = false;
            designer.Caption = "Chart Designer";
            designer.EnableLargeDataSetWarning = true;

            designer.ChartElementHighlighting += OnChartElementHighlighting;
            designer.ChartElementSelecting += OnChartElementSelecting;
            designer.ChartStructureUpdating += OnChartStructureUpdating;

            designer.ShowDialog();
            #endregion #DesignerCustomization
        }

        #region #BehaviorEvents
        // The ChartStructureUpdating event allows for modifying chart model.
        private void OnChartStructureUpdating(object sender, ChartStructureChangingEventArgs e) {
            e.ChartModel.Series.AllowAddChild = false;
            foreach(SeriesModel seriesModel in e.ChartModel.Series) { 
                seriesModel.AllowChangeVisibility = false;
                seriesModel.AllowRemove = false;
            }
            e.ChartModel.Legends.ShowInStructureControl = false;
        }

        // The ChartElementSelecting event allows for preventing element selection or
        // modifying the selecting element edit controls.
        private void OnChartElementSelecting(object sender, ChartElementSelectingEventArgs e) {
            if(e.ElementModel is SeriesBaseModel) {
                e.Cancel = true;
                return;
            }
            if (e.ElementModel is LegendModel) {
                e.ShowPropertiesTab = false;
                e.ShowDataTab = false;
                e.CustomOptionsControl = new CustomLegendOptionsControl(e.ElementModel); 
            }
        }

        // The ChartElementSelecting event allows for preventing element highlight.
        private void OnChartElementHighlighting(object sender, ChartElementHighlightingEventArgs e) {
            SeriesBaseModel series = e.ElementModel as SeriesBaseModel;
            if(series == null) {
                e.Cancel = true;
                return;
            }

        }
        #endregion #BehaviorEvents
    }
}
